### PR TITLE
Fix backend not being set for setting up serving params

### PR DESCRIPTION
**Pull request by danmcp**
With the previous logic, the backend was actually selected after the params were setup.  This new logic makes sure we have the right backend selected before params are configured.
    

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the
  [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [ ] [Changelog](https://github.com/instructlab/instructlab/blob/main/CHANGELOG.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: src/instructlab/model/evaluate.py [hunk 1]
```diff
@@ -248,7 +248,8 @@ def launch_server(
 
     eval_serve = copy(ctx.obj.config.serve)
     if backend is None:
-        eval_serve.backend = backends.VLLM
+        backend = eval_serve.backend = backends.VLLM
```
Comment (nathan-weinberg) [inline src/instructlab/model/evaluate.py:251]: Just to make sure I'm understanding this correctly - this is setting `backend` and `eval_serve.backend` vars to the value of `backends.VLLM`?

Comment (danmcp) [reply src/instructlab/model/evaluate.py:251]: Correct

**Review by nathan-weinberg — APPROVED** [review]

File: src/instructlab/model/evaluate.py (hunk 1 above)
Comment (danmcp) [reply src/instructlab/model/evaluate.py]: Update, I've changed the logic to figure out the backend with backends.  That resolves the other issue of potentially setting up params for the wrong type.

**Review by cdoern — APPROVED** [review]
lgtm nice catch.

if we funnel this from the cfg, then this won't be an issue I think

**Review by alinaryan — APPROVED** [review]

**Comment by cdoern** [comment]
macos is flaky, lets keep rerunning until is passes

**Review by alimaredia — APPROVED** [review]
In the absence of CI I ran MT-Bench and MT-Bench-Branch by hand on a g5 instance and the commands ran as expected.

**Pull request merged**